### DISCUSSION TITLE
Trying to replace lineChart with new chart

### DISCUSSION THREAD

**Discussion by matsandreassen-triona** [primefaces/community#2260] [PrimeFaces] [answered]
After upgrading from 13 to 14, I decided to attempt upgrading an old line chart to the new chart component.

I consulted the showcase at https://www.primefaces.org/showcase/ui/chart/line.xhtml and proceeded to try adapting the code but only one of the classes in the sample code resolve at all. Only org.primefaces.component.linechart.LineChart resolves and that doesn't even have a toJson() method.

#AmITheIdiot

**Comment by matsandreassen-triona** [comment, marked as answer]
Doh!

So, there's an XDEV Java dependency for this, and there was a sentence about it at the top of the showcase page. Yes, I'm the idiot.

I still have a suggestion though: maybe indicate this in the manual under "dependencies"?

**Reply by melloware** [reply]
We were going to but since its NOT a required dependencies its only a suggestion we did not want to list it.  You can technically generate the JSON for CHarts JS any way you want from raw string to GSON to of course the nice XDEV library!  But at the end of the day the component just accepts the raw JSON however you get there.
